test/e2e/network/netpol/network_policy.go

```diff
@@ -119,21 +119,27 @@ and what is happening in practice:
 
 var _ = common.SIGDescribe("Netpol", func() {
 	f := framework.NewDefaultFramework("netpol")
+	//modelMap maps the root namespace of the test to the model created before running the test.
+	var modelMap = make(map[string]*Model)
```

Comment:
There are multiple goroutines reading from and writing to `modelMap`, right?  This looks like a concurrency bug.

Reply:
why do we need a map to share across the tests?
you only need a model by test, no? 
you can create just global `var model  Model` and initialize it in BeforeEach, it is a closure variable 
https://onsi.github.io/ginkgo/#extracting-common-setup-beforeeach

Reply:
yes, that was the original idea but we noticed issues while running the tests in parallel. `model` was being shared by multiple tests and there would only one root namespace for multiple tests.

Reply:
... pfff, ginkgo is very opinionated, we should have this very clear or we'll end with unpredictable results...
each `It.` block must be independent  https://onsi.github.io/ginkgo/#avoiding-test-pollution

Reply:
so keeping `var model Model` global and initializing it in BeforeEach should not be issue even if tests are run in parallel?

Reply:
global as in the `Describe` block, you have more details here https://github.com/onsi/ginkgo/issues/440

Reply:
imo we could just skip relying on `BeforeEach`  entirely, and initialize inside of the individual tests in a one liner function that send framework as an arg to a `initializeNamespacePods` or whatever, if we dont want to play the ginkgo guessing game.

Reply:
Now we have 3 options:
1. stick to the modelMap approach but that may have concurrency issues
2. go with global model in `Describe` block and initialize it in `BeforeEach`
3. what Jay suggested

I have tried 1 and 2 locally with 2 ginkgo nodes, running tests in parallel and it worked for both. I don't see a reason why Jay's approach will fail either, because it is same as 2, has the same lines of `BeforeEach` in every `It`.

Let me know what we should go with.

Reply:
If we are under ginkgo we should do as ginkgo recommend, if all options are equivalent , option 2 is what ginkgo documents
https://github.com/kubernetes/kubernetes/pull/102354#discussion_r667853246

Reply:
cool, I'll add the commit